quartz-core/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java

```diff
@@ -1598,6 +1598,37 @@ public TriggerState getTriggerState(Connection conn, TriggerKey key)
 
     /**
      * <p>
+     * Reset the current state of the identified <code>{@link Trigger}</code>.
+     * </p>
+     *
+     * @see TriggerState#NORMAL
+     * @see TriggerState#PAUSED
+     * @see TriggerState#COMPLETE
+     * @see TriggerState#ERROR
+     * @see TriggerState#NONE
+     */
+    public void resetTriggerState(final TriggerKey triggerKey) throws JobPersistenceException {
+        executeWithoutLock(
+                new TransactionCallback() {
+                    public Object execute(Connection conn) throws JobPersistenceException {
+                        resetTriggerState(conn, triggerKey);
+                        return null;
+                    }
+                });
+    }
+
+    public void resetTriggerState(Connection conn, TriggerKey key)
+            throws JobPersistenceException {
+        try {
+            getDelegate().updateTriggerState(conn, key, STATE_WAITING);
```

Comment:
I also wonder if this should only work for triggers in ERROR state.   (vs.  acquired, blocked, paused).